src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -1236,4 +1249,90 @@ public ParticipationStatus getParticipationStatus( UUID studyId, String particip
+                                .isPresent() )
+                        .filter( neighbor -> neighbor.getNeighborId().get().toString()
+                                .equals( questionnaireEKID.toString() ) )
+                        .map( neighbor -> neighbor.getNeighborDetails().get() )
+                        .findFirst() // If a study has multiple questionnaires, we are only interested in the one with a matching EKID
+                        .ifPresent( questionnaire::setQuestionnaireDetails );
+
+                if ( questionnaire.getQuestionnaireDetails() == null ) {
+                    logger.info( "questionnaire does not exist - studyId: {}, questionnaireEKID: {}, neighbors: {}",
+                            studyId,
+                            questionnaireEKID,
+                            neighbors );
+                    throw new IllegalArgumentException(
+                            "questionnaire does not exist, studyId: " + studyId + "questionnaire EKID = "
+                                    + questionnaireEKID );
+                }
+                logger.info( "retrieved questionnaire: {}", questionnaire.getQuestionnaireDetails().toString() );
+
+                // get questions neighboring questionnaire
+                neighbors = searchApi.executeFilteredEntityNeighborSearch(
+                        questionnaireESID,
+                        new EntityNeighborsFilter(
+                                Set.of( questionnaireEKID ),
+                                java.util.Optional.of( Set.of( questionsESID ) ),
+                                java.util.Optional.of( Set.of( questionnaireESID ) ),
+                                java.util.Optional.of( Set.of( partOfESID ) )
+                        )
+                );
+
+                List<Map<FullQualifiedName, Set<Object>>> questions = neighbors
+                        .getOrDefault( questionnaireEKID, List.of() )
+                        .stream()
+                        .filter( neighbor -> neighbor.getNeighborDetails().isPresent() )
+                        .map( neighbor -> neighbor.getNeighborDetails().get() )
+                        .collect( Collectors.toList() );
+
+                questionnaire.setQuestions( questions );
+
+                logger.info( "retrieved {} questions associated with questionnaire {}",
+                        questions.size(),
+                        questionnaireEKID );
+
+                return questionnaire;
+            }
+        } catch ( Exception e ) {
+            // catch all errors encountered during execution
+            logger.error( "unable to retrieve questionnaire: studyId = {}, questionnaire = {}",
+                    studyId,
+                    questionnaireEKID );
+            throw new RuntimeException( "caught exception", e );
+        }
+        throw new IllegalArgumentException( "questionnaire not found: questionnaire =" + questionnaireEKID );
```

Comment:
I think this exception make it back to the caller. Keep it simple for now, just "questionnaire not found".

Reply:
Good point!